tests/rptest/tests/random_node_operations_test.py

```diff
@@ -0,0 +1,408 @@
+                   timeout_sec=self.timeout,
+                   backoff_sec=1)
+
+    def add(self, idx: int):
+        self.logger.info(f"executor - adding node (idx: {idx})")
+
+        node = self.redpanda.get_node(idx)
+
+        self.redpanda.start_node(node,
+                                 timeout=self.timeout,
+                                 auto_assign_node_id=True,
+                                 omit_seeds_on_idx_one=False)
+
+        self.logger.info(
+            f"added node: {idx} with new node id: {self.node_id(idx)}")
+
+    def _replicas_per_node(self):
+        kafkacat = KafkaCat(self.redpanda)
+        node_replicas = defaultdict(int)
+        md = kafkacat.metadata()
+        for topic in md['topics']:
+            for p in topic['partitions']:
+                for r in p['replicas']:
+                    id = r['id']
+                    node_replicas[id] += 1
+
+        return node_replicas
+
+    def wait_for_rebalanced(self, idx: int):
+
+        node_id = self.node_id(idx)
+        self.logger.info(
+            f"executor - waiting for  node {node_id} (idx: {idx}) to be rebalanced"
+        )
+
+        def has_new_replicas():
+            per_node = self._replicas_per_node()
+            self.logger.info(
+                f"waiting for node {node_id} replicas, current replicas per node: {per_node}"
+            )
+            return node_id in per_node and per_node[node_id] > 5
+
+        wait_until(has_new_replicas, timeout_sec=self.timeout, backoff_sec=1)
+
+    def execute_operation(self, operation: NodeOperation):
+        start = time.perf_counter()
+        self.logger.info(
+            f"executor - node {operation.node} operation: {operation.type}, wait for finish: {operation.wait_for_finish}"
+        )
+        if operation.type == OperationType.ADD:
+            self.add(operation.node)
+            if operation.wait_for_finish:
+                self.wait_for_rebalanced(operation.node)
+        elif operation.type == OperationType.DE_RECOMMISSION:
+            self.decommission(operation.node)
+
+            self.recommission(operation.node)
+            if operation.wait_for_finish:
+                self.wait_for_rebalanced(operation.node)
+        elif operation.type == OperationType.DECOMMISSION:
+            # query node_id before removing a node
+            node_id = self.node_id(operation.node)
+            self.decommission(operation.node)
+            if operation.wait_for_finish:
+                self.wait_for_removed(node_id)
+                self.stop_node(operation.node)
+            else:
+                self.stop_node(operation.node)
+                self.wait_for_removed(node_id)
+        self.logger.info(
+            f"executor - node {operation.node} operation: {operation.type}, wait for finish: {operation.wait_for_finish}, took {time.perf_counter() - start} seconds"
+        )
+
+
+class RandomNodeOperationsTest(EndToEndTest):
```

Comment:
It's not obvious why this is something separate from `node_operations_fuzzy_test.py` -- what's the conceptual difference?

Reply:
The idea is to run simplified version of nodes operations fuzzy test before every merge. I will create a follow up PR in which i will reuse the common parts.